osi_environment.proto

```diff
@@ -324,4 +233,102 @@ message EnvironmentalConditions
         //
         optional uint32 seconds_since_midnight = 1;
     }
+
+    //
+    // \brief Definition of the ambient illumination
+    //
+    // \note Ambient light is any light in the vehicle's environment that is not
+    // emitted by the vehicle itself. It can include sun/moon light, light from
+    // other cars, street lights etc.
+    //
+    message AmbientIllumination
+    {
+        // Definition of discretized ambient illumination states:
+        //
+        // lx ("lux") is the SI unit of luminance or illumination of an area of exact
+        // one square meter, which is equal to one lumen per square meter 1 lx =
+        // 1 lm/m^2 [1].
+        // lm ("lumen") is the SI derived unit of luminous flux and a measure of
+        // the total quantity of visible light emitted by a source. The lumen is
+        // defined in relation to cd ("candela") as 1 lm = 1 cd sr, where sr
+        // denotes steradian, the unit of solid angle used in 3D geometry analogous
+        // to the radian [1].
+        //
+        // Categorization is done based on natural day/night time illuminance levels
+        // [2] and standards for required lighting levels on roads [2, 3, 4, 5].
+        //
+        // \par References:
+        // [1] DIN Deutsches Institut fuer Normung e. V. (1982). <em>DIN 5031-3 Strahlungsphysik im optischen Bereich und Lichttechnik - Groessen, Formelzeichen und Einheiten der Lichttechnik</em>. (DIN 5031-3:1982-03). Berlin, Germany. \n
+        // [2] National Optical Astronomy Observatory. (2015, December 02). <em>Recommended Light Levels</em>. Retrieved January 25, 2020, from https://www.noao.edu/education/QLTkit/ACTIVITY_Documents/Safety/LightLevels_outdoor+indoor.pdf \n
+        // [3] Wang, Y. & Zou, Y., (2016, March). <em>Study on Illumination for State Highways</em>. Washington State Department of Transportation. Retrieved January 25, 2020, from http://www.wsdot.wa.gov/research/reports/fullreports/847.1.pdf \n
+        // [4] Laperriere, A. (2011, May). <em>LED street lighting in the municipality of Saint-Gedeon-de-Beauce within the framework of advanced lighting technologies</em>. Retrieved January 25, 2020, from http://sslnet.ca/wp-content/uploads/2011/10/LTE-RT-2011-0076-Anglais.pdf \n
+        // [5] Crabb, G. I., Beaumont, R. & Webster, D. (2008, October 17). <em>Review of the class and quality of street lighting</em>. Transport Research Laboratory. Retrieved January 25, 2020, from http://courtneystrong.com/wp-content/uploads/2017/07/css-sl1-class-and-quality-of-street-lighting.pdf
+        //
+        enum AmbientIlluminationLevel
+        {
+            // Ambient illumination is unknown (must not be used in ground truth).
+            //
+            AMBIENT_ILLUMINATION_UNKNOWN = 0;
```

Comment:
The enum types should be prefixed with the full enum name and include another underscore right before the actual type. So if the enum name is `AmbientIlluminationLevel` it should be `AMBIENT_ILLUMINATION_LEVEL_1`, `AMBIENT_ILLUMINATION_LEVEL_OTHER `etc.

see https://opensimulationinterface.github.io/osi-documentation/#_interface_conventions

Reply:
Right, but then the naming goes like this
`AMBIENT_ILLUMINATION_LEVEL_UNKNOWN = 0`
...
`AMBIENT_ILLUMINATION_LEVEL_LEVEL1`
`AMBIENT_ILLUMINATION_LEVEL_LEVEL2`
sounds wired, how to solve this?

Reply:
`AMBIENT_ILLUMINATION_LEVEL_1` would not sound too bad, I think. Though, I don't know if we actually need the "AmbientIllumination" term in the types if the enum is located within the `AmbientIllumination` message anyway. Maybe we could name the enum `Level` and have the types `LEVEL_UNKOWN`, `LEVEL_OTHER`, `LEVEL_1`...`LEVEL_9`? It has been done similarly with `Type`, `Material`, etc. in other messages like `StationaryObject`.